.github/workflows/build-push.yml

```diff
@@ -0,0 +1,155 @@
+            debian-image:   "buster-slim"
+            debian-version: "10"
+            aliases: |
+              quay.io/evl.ms/fullstaq-ruby:2.6.6-jemalloc-slim
+              quay.io/evl.ms/fullstaq-ruby:2.6-jemalloc-slim
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "malloctrim"
+            debian-image:   "buster"
+            debian-version: "10"
+            aliases:  |
+              quay.io/evl.ms/fullstaq-ruby:2.6.6-malloctrim
+              quay.io/evl.ms/fullstaq-ruby:2.6-malloctrim
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "malloctrim"
+            debian-image:   "buster-slim"
+            debian-version: "10"
+            aliases: |
+              quay.io/evl.ms/fullstaq-ruby:2.6.6-malloctrim-slim
+              quay.io/evl.ms/fullstaq-ruby:2.6-malloctrim-slim
+
+          # 2.6.6 on Debian 9
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "jemalloc"
+            debian-image:   "stretch"
+            debian-version: "9"
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "jemalloc"
+            debian-image:   "stretch-slim"
+            debian-version: "9"
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "malloctrim"
+            debian-image:   "stretch"
+            debian-version: "9"
+          - ruby-version:   "2.6.6"
+            ruby-variant:   "malloctrim"
+            debian-image:   "stretch-slim"
+            debian-version: "9"
+
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          install: true
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-dev-${{ matrix.debian-image }}-${{ matrix.ruby-version }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-dev-${{ matrix.debian-image }}-${{ matrix.ruby-version }}
+            ${{ runner.os }}-buildx-dev-${{ matrix.debian-image }}
+            ${{ runner.os }}-buildx-dev
+      - name: Login to Quay.io
+        if:   github.event_name != 'pull_request'
+        uses: docker/login-action@v1 
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }} # Get it at Settings → CLI Password
+      - name: Build Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile
+          pull: true
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: |
+            quay.io/evl.ms/fullstaq-ruby:${{ matrix.ruby-version }}-${{ matrix.ruby-variant }}-${{ matrix.debian-image }}
+            ${{ matrix.aliases }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
```

Comment:
is there a reason why you guys don't do the cache dance with the run of the `mv new_cache_location established_cache_location`, as suggested in [the official documentation](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#local-cache) - and even in [your articles](https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching#the-cache-dance-off) (note: I realise this PR predates the article, but since it does so by only one month, I assume the article to still be relevant)(?)
(asking because I'm genuinely curious if there's something else at play here that make the use of that workaround irrelevant)
@Envek